lib/management/configuration.js

```diff
@@ -116,6 +116,19 @@ function patchConfiguration(newConf, log, cb) {
                     break;
                 case 'location-scality-ring-s3-v1':
                     pathStyle = true; // fallthrough
+                    let https = false;
+                    let ringEndpoint = l.details.endpoint ||
+                        's3.amazonaws.com';
+                    if (ringEndpoint.includes('://')) {
+                        const url = new URL(awsEndpoint);
+                        ringEndpoint = url.host;
+                        https = url.scheme === 'https';
+                    }
+                    location.details = {
+                        awsEndpoint: ringEndpoint,
```

Comment:
missing properties? `credentials`, `bucketName`, `bucketMatch`, sse, `supportsVersioning`, `pathStyle`, `https`.

Reply:
Are these set by default somewhere if nothing is specified for location.details? Cos what is being set doesn't seem consistent across the aws-derived locations

Reply:
They were set in the fallthrough code path

Reply:
the defaults are empty.